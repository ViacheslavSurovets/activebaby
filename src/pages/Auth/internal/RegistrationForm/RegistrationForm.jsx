import React, { useState, useMemo } from 'react';
import { connect } from 'react-redux';
import { useTranslation } from 'react-i18next';

import { CustomButton, CustomForm, CustomInput } from '@components';
import {
  LoginContainerButtonWrapper,
  LoginContainerTitle,
  LoginContentWrapper,
  LoginNewCustomerContainer,
  LoginNewCustomerLink,
  LoginNewCustomerText
} from '@pages/Auth/styles';

import { signInWithGoogle, auth, createUserProfileDocument } from '@core/firebase';

import { signUpStart } from '@redux/user/user.actions';

import PropTypes from 'prop-types';

const RegistrationForm = ( { signUpStart } ) => {

  const [ userCredentials, setUserCredentials ] = useState ( {
    firstName: '',
    lastName: '',
    email: '',
    password: '',
    confirmPassword: '',
    checkbox: true
  } );

  const { t } = useTranslation ();

  const { firstName, lastName, email, password, confirmPassword, checkbox } = userCredentials;

  const handleSubmit = async ( event ) => {
    event.preventDefault ();
    if ( password !== confirmPassword ) {
      alert ( t ( 'authPage.checkPasswords' ) );
      return;
    }

    await signUpStart ( { firstName, lastName, email, password } );

    setUserCredentials ( {
      firstName: '',
      lastName: '',
      email: '',
      password: '',
      confirmPassword: '',
      checkbox: true
    } );
  };

  const handleChange = ( event ) => {
    const { value, name } = event.target;
    setUserCredentials ( { ...userCredentials, [ name ]: value } );
  };

  const dataRegistrationForm = useMemo ( () => {
    return [
      {
        type: 'text',
        name: 'firstName',
        value: firstName,
        autocomplete: 'on',
        label: 'authPage.firstName',
      },
      {
        type: 'text',
        name: 'lastName',
        value: lastName,
        autocomplete: 'on',
        label: 'authPage.lastName',

      },
      {
        type: 'email',
        name: 'email',
        value: email,
        autocomplete: 'on',
        label: 'authPage.email',
      },
      {
        type: 'password',
        name: 'password',
        value: password,
        autocomplete: 'current-password',
        label: 'authPage.password',
      },
      {
        type: 'password',
        name: 'confirmPassword',
        value: confirmPassword,
        label: 'authPage.confirmPassword',
      },
      {
        type: 'checkbox',
        name: 'checkbox',
        value: { checkbox },
        label: null
      }
    ];
  }, [ firstName, email, lastName, password, confirmPassword ] );

  return (
    <>
      <LoginContainerTitle>{ t ( 'authPage.registrationTitle' ) }</LoginContainerTitle>
      <LoginContentWrapper>
        <CustomForm onSubmit={ handleSubmit }>
          {
            dataRegistrationForm.map ( ( item, idx ) =>
              <CustomInput
                key={ idx }
                type={ item.type }
                name={ item.name }
                value={ item.value }
                label={ t ( item.label ) }
                autocomplete={ item.autocomplete }
                onChange={ handleChange }
                inlineBlock
                required
                big
              />
            )
          }
          <LoginContainerButtonWrapper>
            <CustomButton
              type='submit'
              orangeSoft
              marginTop={ .5 }
              width={ '40%' }
              mobile
            >
              { t ( 'authPage.registrationButton' ) }
            </CustomButton>
            <CustomButton
              onClick={ signInWithGoogle }
              marginTop={ .5 }
              width={ '40%' }
              googleButton
              mobile
            >
              { t ( 'authPage.enterWithGoogle' ) }
            </CustomButton>
          </LoginContainerButtonWrapper>
        </CustomForm>
        <LoginNewCustomerContainer>
          <LoginNewCustomerText>{ t ( 'authPage.registrationQuestion' ) }</LoginNewCustomerText>
          <LoginNewCustomerLink
            to='/auth'
          >{ t ( 'authPage.loginLink' ) }</LoginNewCustomerLink>
        </LoginNewCustomerContainer>
      </LoginContentWrapper>
    </>
  );
};


const mapDispatchToProps = dispatch => ({
  signUpStart: ( user ) => dispatch ( signUpStart ( user ) )
});


export default connect ( null, mapDispatchToProps ) ( RegistrationForm );

RegistrationForm.propTypes = {
  signUpStart: PropTypes.func
};
