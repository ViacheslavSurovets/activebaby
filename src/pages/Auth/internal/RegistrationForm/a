import React from 'react';
import { useTranslation } from 'react-i18next';
import { InputWrapper } from '@src/styles/commonStylesAndStyledComponents';
import { CustomButton, CustomForm, CustomInput, CustomLabel } from '@components';
import {
  LoginContainerButtonWrapper,
  LoginContainerTitle,
  LoginContentWrapper,
  LoginNewCustomerContainer,
  LoginNewCustomerLink,
  LoginNewCustomerText
} from '@pages/Auth/styles';

const RegistrationForm = ( props ) => {
  const { t } = useTranslation ();

  return (
    <>
      <LoginContainerTitle>{ t ( 'loginPage.registrationTitle' ) }</LoginContainerTitle>
      <LoginContentWrapper>
        <CustomForm>
          <InputWrapper>
            <CustomInput inlineBlock big />
            <CustomLabel>{ t ( 'loginPage.firstName' ) }</CustomLabel>
          </InputWrapper>
          <InputWrapper>
            <CustomInput inlineBlock big />
            <CustomLabel>{ t ( 'loginPage.lastName' ) }</CustomLabel>
          </InputWrapper>
          <InputWrapper>
            <CustomInput inlineBlock big />
            <CustomLabel>{ t ( 'loginPage.login' ) }</CustomLabel>
          </InputWrapper>
          <InputWrapper>
            <CustomInput inlineBlock big />
            <CustomLabel>{ t ( 'loginPage.password' ) }</CustomLabel>
          </InputWrapper>
          <LoginContainerButtonWrapper>
            <CustomButton orangeSoft>{ t ( 'loginPage.registrationButton' ) }</CustomButton>
          </LoginContainerButtonWrapper>
        </CustomForm>
        <LoginNewCustomerContainer>
          <LoginNewCustomerText>{ t ( 'loginPage.registrationQuestion' ) }</LoginNewCustomerText>
          <LoginNewCustomerLink
            to='/auth'
          >{ t ( 'loginPage.loginLink' ) }</LoginNewCustomerLink>
        </LoginNewCustomerContainer>
      </LoginContentWrapper>
    </>
  );
};

export default RegistrationForm;
