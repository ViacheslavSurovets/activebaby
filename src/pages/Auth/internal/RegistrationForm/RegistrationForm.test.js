import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import RegistrationForm from './RegistrationForm';
import configureMockStore  from 'redux-mock-store';


const mockStore = configureMockStore();
const store = mockStore({});

describe('RegistrationForm', ()=> {
  it('RegistrationForm Snapshot', ()=> {
    expect ( shallow (
      <Provider store={store}>
        <RegistrationForm  />
      </Provider>
    ) ).toMatchSnapshot();
  });
});
