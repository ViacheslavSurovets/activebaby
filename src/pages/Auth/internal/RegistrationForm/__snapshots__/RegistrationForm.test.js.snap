// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm RegistrationForm Snapshot 1`] = `ShallowWrapper {}`;
